backend/src/charityprojects/api/views.py

```diff
@@ -895,3 +897,100 @@ def get_challenge_learn_new_skill(request, project_id, user_email):
+                prize_id = project_user_details_record.prize_given_id
+                prize_details = Prize.objects.filter(id=prize_id).first()
+                if prize_details:
+                    response['image'] = request.build_absolute_uri(prize_details.image.url)
+            adventure_id = project_user_record.adventure_id
+            response['adventure_id'] = adventure_id
+            if adventure_id == 1:
+                invitees = []
+                challenge_spread_word = SpreadWord.objects.filter(project_user_id=pu_id).first()
+                if challenge_spread_word:
+                    spread_word_pu_id = challenge_spread_word.project_user_id
+                    unregister_invitation = UnregisterInvitation.objects.filter(project_user_id=spread_word_pu_id).values()
+                    if unregister_invitation:
+                        for item in unregister_invitation:
+                            invitees.append(item['unregister_user_emailId'])
+                    if spread_word_pu_id == pu_id:
+                        registered_invitation = UserInvitation.objects.filter(project_id=project_id, user_id=user_id).values()
+                        if registered_invitation:
+                            for item in registered_invitation:
+                                user = User.objects.get(id=item['friend_id'])
+                                invitees.append(user.email)
+                    response['invitees'] = invitees
+                    if project_user_details_record.video:
+                        response['video'] = request.build_absolute_uri(project_user_details_record.video.url)
+                    else:
+                        response['video'] = ''
+            elif adventure_id == 2:
+                challenge_skill = LearnNewSkill.objects.filter(project_user_id=pu_id).first()
+                if challenge_skill:
+                    response['new_skill'] = challenge_skill.new_skill
+                    response['description'] = challenge_skill.description
+                    if challenge_skill.video:
+                        response['video'] = request.build_absolute_uri(challenge_skill.video.url)
+                    else:
+                        response['video'] = ''
+            elif adventure_id == 3:
+                challenge_develop_habit = DevelopNewHabit.objects.filter(project_user_id=pu_id).first()
+                if challenge_develop_habit:
+                    response['new_habit'] = challenge_develop_habit.new_habit
+                    response['description'] = challenge_develop_habit.description
+                    if challenge_develop_habit.video:
+                        response['video'] = request.build_absolute_uri(challenge_develop_habit.video.url)
+                    else:
+                        response['video'] = ''
+            elif adventure_id == 4:
+                challenge_voluteer_time = VolunteerTime.objects.filter(project_user_id=pu_id).first()
```

Comment:
Can we use the serializers here get the details for the response?

Reply:
That's a good suggestion. We can try implementing that in the next sprint.